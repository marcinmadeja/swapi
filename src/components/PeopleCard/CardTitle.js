import React from 'react';
import PersonIcon from 'material-ui-icons/Person';

import {
  Title,
  TitlePicture,
  UserName,
} from './CardTitle.styles';

const CardTitle = ({ name }) => {
  return (
    <Title>
      <TitlePicture>
        <PersonIcon />
      </TitlePicture>

      <UserName>{name}</UserName>
    </Title>
  );
};

export default CardTitle;
