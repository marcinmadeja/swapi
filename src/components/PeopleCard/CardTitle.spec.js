import React from 'react';
import 'jest-styled-components';
import CardTitle from './CardTitle';

describe('CardTitle', () => {
  it('should render element', () => {
    const component = shallow(<CardTitle name="" />);
    expect(component.exists()).toBe(true);
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(<CardTitle name="user name" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
