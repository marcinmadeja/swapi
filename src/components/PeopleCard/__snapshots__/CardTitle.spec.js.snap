// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 60px;
  height: 60px;
  margin: 0 auto;
  border-radius: 100%;
  background: rgba(0,0,0,0.04);
  color: rgba(0,0,0,0.3);
}

.c1 svg {
  width: 30px;
  height: 30px;
}

.c2 {
  margin: 15px 0 0;
  font-size: 18px;
  font-weight: 500;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <h4
    className="c2"
  >
    user name
  </h4>
</div>
`;
